3.0-canonical-k8s (#931)
spectrocloud/librarium: content/docs/06-integrations/00-canonical-k8s.md

---
title: "Canonical Micro Kubernetes"
metaTitle: "Canonical Micro Kubernetes"
metaDescription: "Canonical Micro Kubernetes pack in Spectro Cloud"
hiddenFromNav: true
isIntegration: true
category: ["kubernetes"]
logoUrl: "https://registry.dev.spectrocloud.com/v1/kubernetes-microk8s/blobs/sha256:b971b64f62e2e67b0a166316f96e6f4211aacea6e28459bb89275e8882ade985?type=image/png"
---

import Tabs from 'shared/components/ui/Tabs';
import WarningBox from 'shared/components/WarningBox';
import InfoBox from 'shared/components/InfoBox';
import PointsOfInterest from 'shared/components/common/PointOfInterest';
import Tooltip from "shared/components/ui/Tooltip";

# Canonical Micro Kubernetes

MicroK8s is a scalable Kubernetes distribution providing core Kubernetes functionalities with optimal resource consumption. 

With limited resource requirements, MicroK8s extends its support in:

* Turning Kubernetes into a lightweight development tool.

* Making Kubernetes available for use in minimal environments such as GitHub CI.

* Adapting Kubernetes for small-appliance IoT applications.

# Prerequisites

* An Ubuntu Operating System

* 20GB Storage

* 4G Memory

> 

Reviewer: The above "prerequisites" seem to come from https://microk8s.io/docs/getting-started. Let's make a reference to their site if reader wants more info.

Author: in references, section added the site

---
title: "Canonical Micro Kubernetes"
metaTitle: "Canonical Micro Kubernetes"
metaDescription: "Canonical Micro Kubernetes pack in Spectro Cloud"
hiddenFromNav: true
isIntegration: true
category: ["kubernetes"]
logoUrl: "https://registry.dev.spectrocloud.com/v1/kubernetes-microk8s/blobs/sha256:b971b64f62e2e67b0a166316f96e6f4211aacea6e28459bb89275e8882ade985?type=image/png"
---

import Tabs from 'shared/components/ui/Tabs';
import WarningBox from 'shared/components/WarningBox';
import InfoBox from 'shared/components/InfoBox';
import PointsOfInterest from 'shared/components/common/PointOfInterest';
import Tooltip from "shared/components/ui/Tooltip";

# Canonical Micro Kubernetes

> MicroK8s is a scalable Kubernetes distribution providing core Kubernetes functionalities with optimal resource consumption. 

Reviewer 2: ```suggestion
Micro Kubernetes is a scalable Kubernetes distribution that provides core Kubernetes functionalities with optimal resource consumption. 
```

> With limited resource requirements, MicroK8s extends its support in:

Reviewer 2: ```suggestion
With limited resource requirements, Micro Kubernetes extends its support with the following:
```

> * Turning Kubernetes into a lightweight development tool.

Reviewer 2: ```suggestion
* Turns Kubernetes into a lightweight development tool.
```

> * Making Kubernetes available for use in minimal environments such as GitHub CI.

Reviewer 2: ```suggestion
* Makes Kubernetes available for use in minimal environments such as GitHub continuous integration (CI).
```

> * Adapting Kubernetes for small-appliance IoT applications.

Reviewer 2: ```suggestion
* Adaps Kubernetes for small-appliance internet of things (IoT) applications.
```

> * Adaps Kubernetes for small-appliance internet of things (IoT) applications.

Reviewer 2: ```suggestion
* Adapts Kubernetes for small-appliance internet of things (IoT) applications.
```

# Prerequisites

* An Ubuntu Operating System

* 20GB Storage

* 4G Memory

* [AmazonEBSCSIDriverPolicy](/clusters/public-cloud/eks#globalroleadditionalpolicies:) to be added to the AWS Cloud Account

# Versions Supported

<Tabs>

<Tabs.TabPane tab="1.23.x" key="1.23.x">

* ** 1.23.4**

</Tabs.TabPane>

</Tabs>

# Quick Workflow

> 1. [Create AWS Cloud Account](/clusters/public-cloud/eks#creatinganawscloudaccount) in Palette Console.

Reviewer 2: ```suggestion
1. [Create an AWS Cloud Account](/clusters/public-cloud/eks#creatinganawscloudaccount) in the Palette Console.
```

> 2. [Create a Infrastructure or Add-on Cluster Profile](/cluster-profiles/task-define-profile) for AWS cloud.

Reviewer 2: ```suggestion
2. [Create an Infrastructure or Add-on Cluster Profile](/cluster-profiles/task-define-profile) for AWS cloud.
```

3. Add Micro Kubernetes as the Kubernetes distribution for the `Infrastructure Layer`.

> 4. Launch your Palette [AWS Cluster](/clusters/public-cloud/aws#deployinganawscluster).

Reviewer 2: ```suggestion
4. Launch Palette [AWS Cluster](/clusters/public-cloud/aws#deployinganawscluster).
```

> 5. Wait for the provisioning to get completed.

Reviewer 2: ```suggestion
5. Wait for provisioning to complete.
```

<InfoBox>
Currently Palette supports Micro Kubernetes for AWS clusters only.
</InfoBox>

## References

> * https://microk8s.io/

Reviewer 3: ```suggestion
- [Micro k8s](https://microk8s.io/)
```

Let's use markdown links whenever possible.

> * https://github.com/canonical/microk8s

Reviewer 3: ```suggestion
- [Canonical Micro k8s](https://github.com/canonical/microk8s)
```
